### PR TITLE
docs/WindowsBuild: Reorder clang-cl & MSVC + Add MSVC Swift invocation

### PR DESCRIPTION
**Pull request by AndrewSB**
Merge after PR #14562

### CODE REVIEW

File: docs/WindowsBuild.md [hunk 1]
```diff
@@ -154,23 +157,39 @@ cmake --build "%swift_source_dir%/build/Ninja-DebugAssert/swift-windows-amd64/ni
...
 
-## Clang-cl
+## MSVC
 
-Follow the instructions for MSVC, but add the following lines to each CMake
-configuration command. `Clang-cl` 4.0.1 has been tested. You can remove the
-`SWIFT_BUILD_DYNAMIC_SDK_OVERLAY=FALSE` definition, as overlays are supported
-with `clang-cl`, as it supports modules. The `Z7` flag is required to produce
-PDB files that MSVC's `link.exe` can read and enables proper stack traces.
+Follow instructions 1-6 for `clang-cl`, but run the following instead to build swift
```
Comment (xwu) [inline docs/WindowsBuild.md]: Nit: uppercase Swift.

**Comment by compnerd** [comment]
The last commit here is excellent!

**Comment by AndrewSB** [comment]
😄 

The last commit is the only one that should show up once #14562 is merged 😉

**Comment by compnerd** [comment]
@AndrewSB please rebase this change

**Comment by AndrewSB** [comment]
@compnerd do the thing

**Comment by compnerd** [comment]
@swift-ci please smoke test and merge

**Comment by AndrewSB** [comment]
What is going on with these infra failures 🤔
@compnerd try again?

**Comment by xwu** [comment]
@swift-ci please smoke test and merge

**Pull request merged**